.github/workflows/ci.yml

```diff
@@ -393,15 +398,20 @@ jobs:
       - uses: ./.github/actions/build-docker
         id: build-docker
         with:
-          # give it a fake name to ensure we never try pushing it
           repository: localhost:5000/camunda/zeebe
           distball: ${{ steps.build-zeebe.outputs.distball }}
-      - name: Verify Docker image
+          platforms: "linux/amd64,linux/arm64"
+          push: true
+      - name: Verify Docker images
         env:
           DATE: ${{ steps.build-docker.outputs.date }}
           REVISION: ${{ github.sha }}
           VERSION: ${{ steps.build-docker.outputs.version }}
-        run: ${PWD}/docker/test/verify.sh '${{ steps.build-docker.outputs.image }}'
+        run: |
+          docker pull --platform linux/amd64 ${{ steps.build-docker.outputs.image }}
```

Comment:
:+1: I'm surprised it's not possible to just do `docker pull --platform amd64,arm64 ...` :shrug:

Reply:
From my current understanding, local docker image tags are always of one particular arch/platform, the actual multi-arch support only exists on the layer of the manifest in the registry. In case multiple images for specific architectures and the same tag exist there are just multiple entries in the manifest of the multi-arch tag ,each pointing to a different digest for the actual image, the client needs to pick one for pulling.

This is also why buildx does not support the `--load` option if you actually build for multiple platforms and forces you to `--push` to a registry. This is the reason I needed to add a local registry service to really test a multi-platform build. 

So docker pull will always pull one image only, the platform may just be implicitly selected if there are multiple. Thus by pulling for different platforms it updates the local tag to point to the platform specific image.

Reply:
I'm still surprised they didn't implement it. There's nothing stopping something like `docker pull --platform arm64,amd64 camunda/zeebe:latest`, and having `docker run camunda/zeebe:latest` just pick the one for the current platform :shrug: Anyway, I was just thinking out loud :smile: